### DISCUSSION TITLE
Diagnosing linker errors

### DISCUSSION THREAD

**Discussion by johngraley** [zephyrproject-rtos/zephyr#55849] [Q&A]
I'm trying to use `console_getline()` in my app. I found `samples/subsys/console/getline/` and saw that it sets 
```
CONFIG_CONSOLE_SUBSYS=y
CONFIG_CONSOLE_GETLINE=y
```
in its `prj.conf`. But when I add the same lines to my `prj.conf` I get linker a error on calling `console_getline()`.

Presumably, the build system has found an unsatisfied dependency or configuration conflict, but instead of simply reporting that and stopping, it's continued as though I had not added `CONFIG_CONSOLE_GETLINE`, leaving the link phase to fail, and a situation that is essentially impossible to debug.

What I want to know is, is there in general a way to debug this? Can I put the configuration tool into a verbose mode where I might then be able to find a message explaining why `CONFIG_CONSOLE_GETLINE=y has been disregarded?

**Comment by yashi** [comment]
Usually, if some configuration is not sutisfied, it prints warnigs at configuration stage:

```
warning: The choice symbol CONSOLE_GETLINE (defined at subsys/console/Kconfig:21) was selected (set
=y), but no symbol ended up as the choice selection. See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_CONSOLE_GETLINE and/or look up
CONSOLE_GETLINE in the menuconfig/guiconfig interface. The Application Development Primer, Setting
Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be helpful
too.
```

This is printed when I do:

```
west build -b nucleo_f429zi zephyr/samples/hello_world -- -DCONFIG_CONSOLE_GETLINE=y
```

with a few lines like `console_getline();` added.

It's obviously missing `-DCONFIG_CONSOLE_SUBSYS=y`.

It'd be helpful to put a [MWE](https://en.wikipedia.org/wiki/Minimal_reproducible_example), if you want to talk about specific case.
